test/spoke/unit/BalanceSheet.t.sol

```diff
@@ -0,0 +1,591 @@
+        _mockEscrowDeposit(erc20, 0, AMOUNT);
+        _mockEscrowDeposit(erc6909, TOKEN_ID, AMOUNT);
+        _mockSendUpdateHoldingAmount(ASSET_20, AMOUNT, ASSET_PRICE, IS_DEPOSIT, !IS_SNAPSHOT, 0);
+
+        vm.startPrank(AUTH);
+        balanceSheet.noteDeposit(POOL_A, SC_1, erc20, 0, AMOUNT);
+        balanceSheet.noteDeposit(POOL_A, SC_1, erc6909, TOKEN_ID, AMOUNT);
+        balanceSheet.submitQueuedAssets(POOL_A, SC_1, ASSET_20);
+
+        (uint128 deposits, uint128 withdrawals) = balanceSheet.queuedAssets(POOL_A, SC_1, ASSET_20);
+        assertEq(deposits, 0);
+        assertEq(withdrawals, 0);
+
+        (,, uint32 queuedAssetCounter, uint64 nonce) = balanceSheet.queuedShares(POOL_A, SC_1);
+        assertEq(queuedAssetCounter, 1);
+        assertEq(nonce, 1);
+    }
+
+    function testSubmitQueuedAssetsTwice() public {
+        _mockSendUpdateHoldingAmount(ASSET_20, AMOUNT, ASSET_PRICE, IS_DEPOSIT, IS_SNAPSHOT, 1);
+
+        vm.startPrank(AUTH);
+        balanceSheet.submitQueuedAssets(POOL_A, SC_1, ASSET_20);
+        balanceSheet.submitQueuedAssets(POOL_A, SC_1, ASSET_20);
+
+        (,,, uint64 nonce) = balanceSheet.queuedShares(POOL_A, SC_1);
+        assertEq(nonce, 2);
+    }
+}
+
+contract BalanceSheetTestSubmitQueuedShares is BalanceSheetTest {
+    function testErrNotAuthorized() public {
+        vm.prank(ANY);
+        vm.expectRevert(IAuth.NotAuthorized.selector);
+        balanceSheet.submitQueuedShares(POOL_A, SC_1);
+    }
+
+    function testSubmitQueuedShares(bool managerOrAuth) public {
+        _mockSendUpdateShares(0, !IS_ISSUANCE, IS_SNAPSHOT, 0);
+
+        vm.prank(managerOrAuth ? MANAGER : AUTH);
+        balanceSheet.submitQueuedShares(POOL_A, SC_1);
+
+        (,,, uint64 nonce) = balanceSheet.queuedShares(POOL_A, SC_1);
+        assertEq(nonce, 1);
+    }
+
+    function testSubmitQueuedSharesWithDeltaPositive() public {
+        _mockShareMint(AMOUNT);
+        _mockSendUpdateShares(AMOUNT, IS_ISSUANCE, IS_SNAPSHOT, 0);
+
+        vm.startPrank(AUTH);
+        balanceSheet.issue(POOL_A, SC_1, RECEIVER, AMOUNT);
+
+        vm.expectEmit();
+        emit IBalanceSheet.SubmitQueuedShares(POOL_A, SC_1, AMOUNT, IS_ISSUANCE, IS_SNAPSHOT, 0);
+        balanceSheet.submitQueuedShares(POOL_A, SC_1);
+
+        (uint128 delta, bool isPositive,, uint64 nonce) = balanceSheet.queuedShares(POOL_A, SC_1);
+        assertEq(delta, 0);
+        assertEq(isPositive, true);
+        assertEq(nonce, 1);
+    }
+
+    function testSubmitQueuedSharesWithDeltaNegative() public {
+        _mockShareBurn(AMOUNT);
+        _mockShareAuthTransferFrom(AUTH, address(balanceSheet), AMOUNT);
+        _mockSendUpdateShares(AMOUNT, !IS_ISSUANCE, IS_SNAPSHOT, 0);
+
+        vm.startPrank(AUTH);
+        balanceSheet.revoke(POOL_A, SC_1, AMOUNT);
+        balanceSheet.submitQueuedShares(POOL_A, SC_1);
+    }
+
+    function testSubmitQueuedSharesAfterUpdateAssets() public {
```

Comment:
Feels like we are missing assertions for this test or should it only ensure that the call order does not revert?